tests/epyccel/recognised_functions/test_numpy_funcs.py

```diff
@@ -3465,3 +3465,362 @@ def create_zeros_like_3_val():
     assert(     f3_shape() ==      create_zeros_like_3_shape()      )
     assert(isclose(     f3_val()  ,      create_zeros_like_3_val()        , rtol=RTOL, atol=ATOL))
     assert(type(f3_val())  == type(create_zeros_like_3_val().item()))
+
+def test_numpy_real_scalar(language):
+
+    @types('bool')
+    def test_bool(a):
+        import numpy as np
+        b = np.real(a)
+        return b
+
+    @types('int')
+    def test_int(a):
+        import numpy as np
+        b = np.real(a)
+        return b
+
+    @types('int8')
+    def test_int8(a):
+        import numpy as np
+        b = np.real(a)
+        return b
+
+    @types('int16')
+    def test_int16(a):
+        import numpy as np
+        b = np.real(a)
+        return b
+
+    @types('int32')
+    def test_int32(a):
+        import numpy as np
+        b = np.real(a)
+        return b
+
+    @types('int64')
+    def test_int64(a):
+        import numpy as np
+        b = np.real(a)
+        return b
+
+    @types('float')
+    def test_float(a):
+        import numpy as np
+        b = np.real(a)
+        return b
+
+    @types('float32')
+    def test_float32(a):
+        import numpy as np
+        b = np.real(a)
+        return b
+
+    @types('float64')
+    def test_float64(a):
+        import numpy as np
+        b = np.real(a)
+        return b
+
+    @types('complex64')
+    def test_complex64(a):
+        import numpy as np
+        b = np.real(a)
+        return b
+
+    @types('complex128')
+    def test_complex128(a):
+        import numpy as np
+        b = np.real(a)
+        return b
+
+    import numpy as np
+
+    bl = np.bool(randint(1e6))
+    integer = randint(1e6)
+    integer8 = np.int8(randint(1e6))
+    integer16 = np.int16(randint(1e6))
+    integer32 = np.int32(randint(1e6))
+    integer64 = np.int64(randint(1e6))
+    fl = np.float(randint(1e6))
+    fl32 = np.float32(randint(1e6))
+    fl64 = np.float64(randint(1e6))
+
+    # gfortran complains about boolean to numeric convertions, given that boolean type in Fortran
+    # is not a numerical type, so Real function in Fortran doesn't accept a non-numerical type in the first argument
+    if (language == 'c'):
+        f_bl = epyccel(test_bool, language=language)
+        assert (f_bl(bl) == test_bool(bl))
```

Comment:
Does this mean that we are missing a call to a casting function in Fortran? Or do you think it is normal that this doesn't work? I'm not even sure what the expected behaviour is so I'm not sure if this is normal or not

Reply:
i just discovered that we can tests them by passing a Literal boolean to the functions.

Reply:
So there is only a problem when passing a boolean variable? That's odd

Reply:
I'm not sure I understood what the problem is/was here

Reply:
Could you explain this further? What was/is the problem?

Reply:
I would like to understand this before this PR is approved/merged and we forget about it

Reply:
after casting arg in PythonReal we dont need to pass a Literal boolean to the functions for the tests, so now we can pass it by a variable.